src/pages/ProvideLiquidity/AddLiquidityForm.tsx

```diff
@@ -0,0 +1,180 @@
+import { useContractKit } from '@celo-tools/use-contractkit'
+import { CELO, ChainId as UbeswapChainId, Token, TokenAmount } from '@ubeswap/sdk'
+import { ButtonError, ButtonPrimary } from 'components/Button'
+import { LightCard } from 'components/Card'
+import { ColumnCenter } from 'components/Column'
+import CurrencyInputPanel from 'components/CurrencyInputPanel'
+import { RowBetween } from 'components/Row'
+import { ApprovalState, useApproveCallback } from 'hooks/useApproveCallback'
+import { PoolPriceBar } from 'pages/AddLiquidity/PoolPriceBar'
+import { Dots } from 'pages/Pool/styleds'
+import React, { useContext } from 'react'
+import { Plus } from 'react-feather'
+import { Field } from 'state/mint/actions'
+import { useDerivedMintInfo, useMintActionHandlers, useMintState } from 'state/mint/hooks'
+import styled, { ThemeContext } from 'styled-components'
+import { TYPE } from 'theme'
+import { maxAmountSpend } from 'utils/maxAmountSpend'
+
+import { ROUTER_ADDRESS } from '../../constants'
+
+const StyledColumnCenter = styled(ColumnCenter)`
+  padding-top: 16px;
+`
+
+interface Props {
+  token0: Token
+  token1: Token
+  onConfirmAddLiquidity: () => void
+}
+
+export default function AddLiquidityForm({ token0, token1, onConfirmAddLiquidity }: Props) {
+  const theme = useContext(ThemeContext)
+
+  const { network } = useContractKit()
+
+  const { independentField, typedValue, otherTypedValue } = useMintState()
+  const {
+    currencies,
+    price,
+    noLiquidity,
+    poolTokenPercentage,
+    dependentField,
+    parsedAmounts,
+    currencyBalances,
+    error,
+  } = useDerivedMintInfo(token0, token1)
+
+  const { onFieldAInput, onFieldBInput } = useMintActionHandlers(noLiquidity)
+
+  // check whether the user has approved the router on the tokens
+  const [approvalA, approveACallback] = useApproveCallback(parsedAmounts[Field.CURRENCY_A], ROUTER_ADDRESS)
+  const [approvalB, approveBCallback] = useApproveCallback(parsedAmounts[Field.CURRENCY_B], ROUTER_ADDRESS)
+
+  const chainId = network.chainId
+
+  // get formatted amounts
+  const formattedAmounts = {
+    [independentField]: typedValue,
+    [dependentField]: noLiquidity ? otherTypedValue : parsedAmounts[dependentField]?.toSignificant(6) ?? '',
+  }
+
+  // get the max amounts user can add
+  const maxAmounts: { [field in Field]?: TokenAmount } = [Field.CURRENCY_A, Field.CURRENCY_B].reduce(
+    (accumulator, field) => {
+      return {
+        ...accumulator,
+        [field]: maxAmountSpend(currencyBalances[field]),
+      }
+    },
+    {}
+  )
+
+  const atMaxAmounts: { [field in Field]?: TokenAmount } = [Field.CURRENCY_A, Field.CURRENCY_B].reduce(
+    (accumulator, field) => {
+      return {
+        ...accumulator,
+        [field]: maxAmounts[field]?.equalTo(parsedAmounts[field] ?? '0'),
+      }
+    },
+    {}
+  )
```

Comment:
```suggestion
  const atMaxAmounts= (field: Field) => maxAmounts[field]?.equalTo(parsedAmounts[field] ?? '0')
```
code golf. didn't test. would need a slight refactor where it's used (replace atMaxAmounts[field] with atMaxAmounts(field)